src/util/loopUnrolling.ml

```diff
@@ -31,34 +31,25 @@ class checkNoBreakVisitor = object
 
 end
 
-let checkNoBreakStmt stmt =
-  let visitor = new checkNoBreakVisitor in
-  ignore @@ visitCilStmt visitor stmt
-
-let checkNoBreakBlock block =
-  let visitor = new checkNoBreakVisitor in
-  ignore @@ visitCilBlock visitor block
-
 class findBreakVisitor(compOption: exp option ref) = object
```

Comment:
Looking at this visitor now, I understand why it didn't want multiple `break`s. It's main goal is to actually find the comparison for the loop, which will be an `if` with a `break` inside.
By only allowing one such `break` it guaranteed that the result is unambiguous: there couldn't be multiple such comparisons found.

Now, if there are multiple matching `if`s with `break`s inside, this will still find the first one because `compOption` is only assigned when it's `None`, so that can happen at most once.
Is the loop condition always in the first matching one though? If so, then why did the old version have a problem at all? Or maybe there are loops where the condition isn't actually the first one?